src/components/Main.tsx

```diff
@@ -60,14 +61,33 @@ export function Main({
   /** Set of supported languages */
   supportedLanguages: ConsentManagerLanguageKey[];
 }): JSX.Element {
+  // need to focus the first button in the modal when the modal is opened
+  const dialogRef = useRef<HTMLDivElement>();
+  useEffect(() => {
+    if (!isViewStateClosed(viewState) && dialogRef.current) {
+      dialogRef?.current?.querySelector('button')?.focus();
+    }
+  }, [viewState, dialogRef]);
+
   // Modal open views
   if (!isViewStateClosed(viewState)) {
     if (!isResponseViewState(viewState)) {
       airgap.setPrompted(true);
     }
     return (
-      <div role="dialog" aria-model="true" className="modal-container">
-        <div role="document" className="modal-container-inner">
+      <div
+        role="dialog"
+        aria-model="true"
+        aria-labelledby="consent-dialog-title"
+        className="modal-container"
+        id="consentManagerMainDialog"
+        ref={
+          dialogRef.current
+            ? (dialogRef as unknown as Ref<HTMLDivElement>)
```

Comment:
cc @blake-transcend i had to do all this to get tsc to be happy...

Reply:
wtf we should always be setting this ref, especially when the current ref is undefined